domain/include/cstone/halos/exchange_halos_gpu.cuh

```diff
@@ -171,6 +186,7 @@ void haloExchangeGpu(int epoch,
     // MUST call MPI_Barrier or any other collective MPI operation that enforces synchronization
     // across all ranks before calling this function again.
     // MPI_Barrier(MPI_COMM_WORLD);
+    return std::tuple<float, float>{timings[0], timings[1]};
```

Comment:
Out of these two values, the first one is wrong and the second one corresponds to the time spent in the entire function which you can observe at the call site. This means you're not getting any added value for this manual instrumentation.